### ISSUE TITLE
Uncaught RangeError: Invalid array length

### ISSUE THREAD

**Issue by Art2Cat** [facebooknuclide/hyperclick#34]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.19.2 x64
**Electron**: 1.6.9
**OS**: Microsoft Windows 10 Enterprise
**Thrown From**: [hyperclick](https://github.com/facebooknuclide/hyperclick) package 0.1.3

### Stack Trace

Uncaught RangeError: Invalid array length

```
At C:\Users\huang.yiming\AppData\Local\atom\app-1.19.2\resources\app\src\text-editor-component.js:2839

RangeError: Invalid array length
    at TextEditorComponent.populateVisibleTiles (~/AppData/Local/atom/app-1.19.2/resources/app/src/text-editor-component.js:2839:45)
    at TextEditorComponent.updateSyncBeforeMeasuringContent (~/AppData/Local/atom/app-1.19.2/resources/app/src/text-editor-component.js:372:16)
    at TextEditorComponent.screenPositionForPixelPosition (~/AppData/Local/atom/app-1.19.2/resources/app/src/text-editor-component.js:2328:18)
    at TextEditorComponent.screenPositionForMouseEvent (~/AppData/Local/atom/app-1.19.2/resources/app/src/text-editor-component.js:1937:23)
    at t.default._getMousePositionAsBufferPosition (/packages/hyperclick/index.js:1:160281)
    at t.default._onMouseMove (/packages/hyperclick/index.js:1:158262)
```

### Commands

```
     -2:32.7.0 intentions:highlight (input.hidden-input)
     -2:32.4.0 core:copy (input.hidden-input)
  9x -1:47.9.0 core:backspace (input.hidden-input)
     -1:37.8.0 intentions:highlight (input.hidden-input)
     -1:37.5.0 core:copy (input.hidden-input)
```

### Non-Core Packages

```
atom-beautify 0.30.5 
atom-material-syntax 1.0.6 
atom-material-ui 2.0.4 
atom-runner 2.7.1 
autoclose-html 0.23.0 
autocomplete-java 1.2.7 
autocomplete-oracle 1.4.1 
autocomplete-xml 0.9.4 
busy-signal 1.4.3 
file-icons 2.1.11 
highlight-selected 0.13.1 
hyperclick 0.1.3 
intellij-idea-keymap 0.2.3 
intentions 1.1.5 
javascript-snippets 1.2.1 
language-log 1.8.0 
language-oracle 1.8.0 
language-pgsql 0.2.3 
linter 2.2.0 
linter-javac 1.9.4 
linter-ui-default 1.6.7 
minimap 4.29.6 
minimap-highlight-selected 4.6.1 
platformio-ide-terminal 2.5.5 
wakatime 7.0.3 
xml-formatter 0.11.0 
```

**Comment by hansonw** [comment]
There've been a couple more fixes to text editor rendering in 1.19.3 and 1.19.4.
Could you upgrade to see if this still repros?

**Comment by hansonw** [comment]
1.19.6 has some more fixes: https://github.com/atom/atom/releases/tag/v1.19.6
Closing because this is a public Atom API.
